File: src/main/java/io/github/binaryfoo/decoders/apdu/VerifyPinAPDUDecoder.java
package io.github.binaryfoo.decoders.apdu;

import io.github.binaryfoo.DecodedData;
import io.github.binaryfoo.decoders.DecodeSession;

public class VerifyPinAPDUDecoder implements CommandAPDUDecoder {
    @Override
    public APDUCommand getCommand() {
        return APDUCommand.Verify;
    }

    @Override
    public DecodedData decode(String input, int startIndexInBytes, DecodeSession session) {
        int length = Integer.parseInt(input.substring(8, 10), 16);
        String data = input.substring(10, 10 + length * 2);
        return DecodedData.primitive("C-APDU: Verify PIN", data, startIndexInBytes, startIndexInBytes + 5 + length);
    }
}

File: src/main/java/io/github/binaryfoo/decoders/apdu/VerifyPinAPDUDecoder.kt
package io.github.binaryfoo.decoders.apdu

import io.github.binaryfoo.DecodedData
import io.github.binaryfoo.decoders.DecodeSession

class VerifyPinAPDUDecoder : CommandAPDUDecoder {
  override fun getCommand(): APDUCommand {
    return APDUCommand.Verify
  }

  override fun decode(input: String, startIndexInBytes: Int, session: DecodeSession): DecodedData {
    val length = Integer.parseInt(input.substring(8, 10), 16)
    val data = input.substring(10, 10 + length * 2)
    return DecodedData(null, "C-APDU: Verify PIN", data, startIndexInBytes, startIndexInBytes + 5 + length)
  }
}
